File: 1927-maximum-ascending-subarray-sum/1927-maximum-ascending-subarray-sum.go
func maxAscendingSum(nums []int) int {
	n := len(nums)
	maxSum := nums[0]
	currSum := nums[0]

	for i := 1; i < n; i++ {
		if nums[i] > nums[i-1] {
			currSum += nums[i]
		} else {
			if currSum > maxSum {
				maxSum = currSum
			}
			currSum = nums[i]
		}
	}
	if currSum > maxSum {
		maxSum = currSum
	}

	return maxSum
}

File: 1927-maximum-ascending-subarray-sum/1927-maximum-ascending-subarray-sum.java
class Solution {
    public int maxAscendingSum(int[] nums) {

        int n = nums.length;
        int max = nums[0];
        int cursum = nums[0];

        for (int i = 1; i < n; i++) {
            if (nums[i - 1] < nums[i]) {
                cursum += nums[i];
            } else {
                if (max < cursum) {
                    max = cursum;
                }
                cursum = nums[i];
            }
        }

        if (cursum > max) {
            max = cursum;
        }

        return max;
    }
}